shared/cluster.go

```diff
@@ -305,6 +305,30 @@ func GetNodes(print bool) ([]Node, error) {
 	return nodes, nil
 }
 
+// GetNodesByRoles takes in one or multiple node roles and returns the slice of nodes that have those roles
+// Valid values for roles are: etcd, control-plane, worker
+func GetNodesByRoles(roles ...string) ([]Node, error) {
+	var nodes []Node
+	var matchedNodes []Node
+
+	for _, role := range roles {
+		cmd := "kubectl get nodes -o wide --sort-by '{.metadata.name}'" +
+			" --no-headers --kubeconfig=" + KubeConfigFile +
+			" -l role-" + role
+		res, err := RunCommandHost(cmd)
+		if err != nil {
+			return nil, err
+		}
+		matchedNodes = append(matchedNodes, parseNodes(res)...)
+	}
+
+	for _, matchedNode := range matchedNodes {
+		nodes = AppendIfMissing(nodes, matchedNode)
```

Comment:
have u created this function only to avoid a nested loop?

Reply:
Not really since the function itself is a loop, this still ends up being a nested loop which sucks. I played around for way too much time with people trying to:
1. Change the kubectl command itself to return everything in one (fails)
2. Use a map (still ends up needing a nested loop)
3. Brainstorm/search for some other solution

This ended up being just the most simple and not too time complex still we're dealing with such small datasets (less than 10 in pretty much all cases)

Reply:
nice, tk you for the explanation